src/main/java/com/almightyalpaca/discord/jdabutler/commands/Dispatcher.java

```diff
@@ -0,0 +1,119 @@
+package com.almightyalpaca.discord.jdabutler.commands;
+
+import com.almightyalpaca.discord.jdabutler.Bot;
+import com.almightyalpaca.discord.jdabutler.commands.commands.*;
+import net.dv8tion.jda.core.entities.TextChannel;
+import net.dv8tion.jda.core.entities.User;
+import net.dv8tion.jda.core.events.message.guild.GuildMessageReceivedEvent;
+import net.dv8tion.jda.core.hooks.ListenerAdapter;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class Dispatcher extends ListenerAdapter {
+
+    private Set<Command> commands = ConcurrentHashMap.newKeySet();
+    private ExecutorService pool = Executors.newCachedThreadPool();
+
+    {
+        registerCommand(new HelpCommand());
+        registerCommand(new VersionsCommand());
+        registerCommand(new Eval());
+        registerCommand(new DocsCommand());
+        registerCommand(new Gradle());
+        registerCommand(new Maven());
+        registerCommand(new Jar());
+        registerCommand(new BuildGradle());
+        registerCommand(new Notify());
+        registerCommand(new Ping());
+        registerCommand(new Uptime());
+        registerCommand(new Changelog());
+    }
+
+    public boolean registerCommand(Command command) {
+        if (command.getName().contains(" ")) {
+            throw new IllegalArgumentException("Name must not have spaces!");
+        }
+        if (commands.stream().map(Command::getName).filter(c -> command.getName().equalsIgnoreCase(c)).count() > 0) {
+            return false;
+        }
+        commands.add(command);
+        return true;
+    }
+
+    public Collection<Command> getCommands() {
```

Comment:
Why is the set not wrapped as immutable and copy of the set?
`return Collections.unmodifiableSet(new HashSet<>(commands);`